lib/spack/spack/bootstrap.py

```diff
@@ -472,3 +634,28 @@ def ensure_clingo_importable_or_raise():
     ensure_module_importable_or_raise(
         module='clingo', abstract_spec=clingo_root_spec()
     )
+
+
+def gnupg_root_spec():
+    # Construct the root spec that will be used to bootstrap clingo
+    spec_str = 'gnupg@2.3:'
+
+    # Add a proper compiler hint to the root spec. We use GCC for
+    # everything but MacOS.
+    if str(spack.architecture.platform()) == 'darwin':
+        spec_str += ' %apple-clang'
+    else:
+        spec_str += ' %gcc'
+
+    target = archspec.cpu.host().family
+    spec_str += ' target={0}'.format(target)
+
+    tty.debug('[BOOTSTRAP ROOT SPEC] gnupg: {0}'.format(spec_str))
+    return spec_str
+
+
+def ensure_gpg_in_path_or_raise():
+    """Ensure gpg or gpg2 are in the PATH or raise."""
+    ensure_executables_in_path_or_raise(
+        executables=['gpg2', 'gpg'], abstract_spec=gnupg_root_spec(),
+    )
```

Comment:
We should think about whether this is sustainable to have a method here for each calling site, or whether these methods should be pushed into the callers.

Reply:
My take on it is the following:
1. The wrapper function may be pushed on the caller (it's indeed just a convenience to fix parameters in one place)
2. I think `gnupg_root_spec` should stay here so that we can collect in the `spack.bootstrap` module all the specs we may need on bootstrapping

Reply:
`gnupg_root_spec` and `clingo_root_spec` are identical methods, besides the spec string. I think we should have a method `get_bootstrap_spec` that takes an abstract spec or abstract spec string as an argument. Then the calling site can keep the information that belongs to it ("I need gpg@1.1:" or whatever it is) and call into bootstrap only to turn that abstract information into a bootstrappable spec.